File: 1063.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split(/\s+/);

let idx = 0;

while (true) {
    const n = parseInt(input[idx++]);
    if (n === 0) break;

    const entrada = [];
    const saida = [];
    const pilha = [];
    let output = [];

    for (let i = 0; i < n; i++)
        entrada.push(input[idx++]);

    for (let i = 0; i < n; i++)
        saida.push(input[idx++]);

    let i = 0, j = 0, flag = 0;

    while (i <= n - 1) {
        pilha.push(entrada[i]);
        output.push('I');
        i++;

        while (pilha.length && j < n && pilha[pilha.length - 1] === saida[j]) {
            pilha.pop();
            output.push('R');
            j++;
            flag++;
        }
    }

    if (flag === n)
        console.log(output.join(''));
    else
        console.log(output.join('') + " Impossible");
}

File: 1063.py
while True:
    n = int(input())
    if n == 0:
        break

    pilha = []
    entrada = input().split()
    saida = input().split()
    output = []

    i = 0
    j = 0
    flag = 0

    while i <= n - 1:
        pilha.append(entrada[i])
        output.append('I')
        i += 1

        while pilha and j < n and pilha[-1] == saida[j]:
            pilha.pop()
            output.append('R')
            j += 1
            flag += 1

    if flag == n:
        print(''.join(output))
    else:
        print(f"{''.join(output)} Impossible")
